src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -926,22 +935,12 @@ public static bool TryCompleteSendTo(SafeSocketHandle socket, ReadOnlySpan<byte>
                         Debug.Assert(buffers == null);
                         sent = SysWrite(socket, buffer, ref offset, ref count, out errno);
                     }
-                    else if (buffers is not null)
-                    {
-                        sent = SysSend(socket, flags, buffers, ref bufferIndex, ref offset, socketAddress, socketAddressLen, out errno);
-                    }
-                    else if (buffer.Length == 0)
-                    {
-                        sent = 0;
-                        errno = Interop.Error.SUCCESS;
-                    }
-                    else if (socketAddress is null)
-                    {
-                        sent = SysSend(socket, flags, buffer, ref offset, ref count, out errno);
-                    }
                     else
                     {
-                        sent = SysSend(socket, flags, buffer, ref offset, ref count, socketAddress, socketAddressLen, out errno);
+                        sent = buffers != null ?
+                            SysSend(socket, flags, buffers, ref bufferIndex, ref offset, socketAddress, socketAddressLen, out errno) :
```

Comment:
Is it possible the buffers list contains a default ArraySegment, or do we screen those out? Wondering if the same issue might exist there.

Reply:
We validate the buffer list when you set it (BufferList setter) and throw on null ArraySegments.